modules/test/playwright/tests/frontend-js-spa-web/formSubmit.spec.ts

```diff
@@ -0,0 +1,119 @@
+import getRandomString from '../../utils/getRandomString';
+import {journalPagesTest} from '../journal-web/fixtures/journalPagesTest';
+import isSPAEnabled from './utils/isSPAEnabled';
+
+export const test = mergeTests(
+	loginTest(),
+	apiHelpersTest,
+	isolatedSiteTest,
+	journalPagesTest,
+	pageEditorPagesTest,
+	productMenuPageTest,
+	pageViewModePagesTest
+);
+
+test(
+	'Assert form submit works correctly on Safari browser',
+	{
+		tag: '@LPD-26285',
+	},
+	async ({
+		apiHelpers,
+		journalEditArticlePage,
+		page,
+		site,
+		widgetPagePage,
+	}) => {
+		const {devices, webkit} = require('playwright');
+		const safari = devices['Desktop Safari'];
+		const layout = await apiHelpers.jsonWebServicesLayout.addLayout({
+			groupId: site.id,
+			options: {
+				type: 'portlet',
+			},
+			title: getRandomString(),
+		});
+
+		await test.step('Naviagate to root page', async () => {
+			await page.goto('/');
+		});
+
+		await test.step('Asset SPA is enabled', async () => {
+			expect(isSPAEnabled({page})).toBeTruthy();
+		});
+
+		await test.step('Add WCD and content with a form including two submit buttons', async () => {
+			await page.goto(`/web${site.friendlyUrlPath}${layout.friendlyURL}`);
+
+			await widgetPagePage.addPortlet('Web Content Display');
+
+			await page
+				.locator(
+					'[id^="_com_liferay_journal_content_web_portlet_JournalContentPortlet_INSTANCE_"]'
+				)
+				.and(page.getByRole('button'))
+				.click();
+
+			await page
+				.getByRole('menuitem', {name: 'Basic Web Content'})
+				.click();
+
+			const title = getRandomString();
+
+			await journalEditArticlePage.fillTitle(title);
+
+			await page.getByLabel('Source', {exact: true}).click();
+
+			await page.locator('.CodeMirror-scroll').click();
+
+			const content =
+				'<form action="" method="post"><input type="submit" value="Button A" /> <input type="submit" value="Button B" /></form><p><span id="capturedFormButtonElement"></span></p><script>Liferay.once("beforeNavigate", function(){buttonValue = Liferay.SPA.__capturedFormButtonElement__.value});Liferay.once("endNavigate", function(){document.getElementById("capturedFormButtonElement").textContent = buttonValue;})</script>';
+
+			await page
+				.getByLabel('Content', {exact: true})
+				.getByRole('textbox')
+				.fill(content);
+
+			await journalEditArticlePage.publishButton.click();
+		});
+
+		await test.step('Set Safari as the browser and check submit buttons in form', async () => {
+			const browser = await webkit.launch();
+
+			const context = await browser.newContext({
+				...safari,
+			});
+
+			const incognitoPage = await context.newPage();
+
+			await incognitoPage.goto(
+				`/web${site.friendlyUrlPath}${layout.friendlyURL}`
+			);
+
+			await incognitoPage.getByRole('button', {name: 'Button B'}).click();
+
+			const capturedFormButtonElement = incognitoPage.locator(
+				'#capturedFormButtonElement'
+			);
+
+			await expect(capturedFormButtonElement).toHaveText('Button B');
```

Comment:
@antonio-ortega I opened a bug for this: [[green-ci] Test for SPA form submission behavior on Safari does not work](https://liferay.atlassian.net/browse/LPD-39675). I'm pinging you as this is related to [On Safari browser, button click events are not registering properly in SPA](https://liferay.atlassian.net/browse/LPD-26285)